src/lib/redux-cozy-api.js

```diff
@@ -0,0 +1,484 @@
+    // { entities: 'io.cozy.photos.albums': { entities: {...}, fetchStatuses: {...}, errors: {...} }}
+    dispatch({ type: FETCH_DOCUMENT, doctype, id })
+    doc = await cozy.client.data.find(doctype, id)
+    // we normalize again...
+    doc = Object.assign({}, doc, { id: doc._id, type: doc._type })
+    dispatch({ type: RECEIVE_DOCUMENT, entity: doc, response: { data: [doc] } })
+  }
+  return doc
+}
+
+export const fetchDocument = (doctype, id, options) => async (dispatch, getState) => {
+  try {
+    const doc = await dispatch(getOrFetchDocument(doctype, id))
+    if (options.include) {
+      const relations = getRelations(getState(), doctype)
+      for (let relationName of options.include) {
+        const relationType = relations[relationName].type
+        // TODO: we only handle files relations here
+        if (relationType === FILES_DOCTYPE) {
+          dispatch(fetchReferencedFiles(doc, relationName))
+        }
+      }
+    }
+  } catch (error) {
+    dispatch({ type: RECEIVE_ERROR, doctype, id, error })
+    console.log(error)
+  }
+}
+
+export const fetchReferencedFiles = (entity, relationName, skip = 0) => async (dispatch, getState) => {
+  try {
+    dispatch({ type: FETCH_REFERENCED_FILES, entity, relationName, skip })
+    // WARN: the stack API is probably not ideal here: referencedFiles are in the 'included' property
+    // (that should be used when fetching an entity AND its relations) and the 'data' property
+    // only contains uplets { id, type }
+    const { included, meta } = await cozy.client.data.fetchReferencedFiles(entity, { skip, limit: FETCH_LIMIT })
+    // we forge a standard response with a 'data' property
+    const response = {
+      data: !included ? [] : included.map(file => Object.assign({}, file.attributes, { id: file.id, _id: file.id, links: file.links })),
+      meta,
+      next: meta.count > skip + FETCH_LIMIT
+    }
+    dispatch({ type: RECEIVE_REFERENCED_FILES, entity, relationName, response, skip })
+  } catch (error) {
+    dispatch({ type: RECEIVE_ERROR, entity, relationName, error })
+    console.log(error)
+  }
+}
+
+export const createEntity = (entity) => async (dispatch, getState) => {
+  try {
+    // TODO: handle this action type
+    dispatch({ type: CREATE_ENTITY, entity })
+    const relations = getRelations(getState(), entity.type)
+    const created = await cozy.client.data.create(entity.type, entity)
+    // Let's handle the entity's relations
+    for (let relationName in relations) {
+      const relationType = relations[relationName].type
+      // TODO: we only handle files relations here
+      if (relationType === FILES_DOCTYPE) {
+        // WARN: we don't use the addReferencedFiles action creator here
+        // because the entity should already have a property named after the
+        // relation's name containing IDs
+        await cozy.client.data.addReferencedFiles(created, entity[relationName])
+      }
+    }
```

Comment:
Shall we factorize the `for relations` loop we see in that function and in `fetchDocument` one's...